src/app/core/import/import-column-mapping/import-column-mapping.component.spec.ts

```diff
@@ -20,50 +15,25 @@ describe("ImportColumnMappingComponent", () => {
     fixture = TestBed.createComponent(ImportColumnMappingComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
-
-    spyOn(component.columnMappingChange, "emit");
-  });
-
-  it("should open mapping component with required data", async () => {
-    component.rawData = [
-      { name: "first", gender: "male" },
-      { name: "second", gender: "female" },
-      { name: "third", gender: "female" },
-    ];
-    component.entityType = TestEntity.ENTITY_TYPE;
-    component.columnMapping = [{ column: "name" }, { column: "gender" }];
-    const openSpy = spyOn(TestBed.inject(MatDialog), "open");
-    openSpy.and.returnValue({ afterClosed: () => of(undefined) } as any);
-
-    const genderColumn = component.columnMapping[1];
-    genderColumn.propertyName = "category";
-    await component.openMappingComponent(genderColumn);
-
-    expect(openSpy).toHaveBeenCalledWith(
-      DiscreteImportConfigComponent,
-      jasmine.objectContaining({
-        data: {
-          col: genderColumn,
-          values: ["male", "female"],
-          entityType: TestEntity,
-        },
-        disableClose: true,
-      }),
-    );
   });
 
-  it("should emit changes after popup is closed", async () => {
-    spyOn(TestBed.inject(MatDialog), "open").and.returnValue({
-      afterClosed: () => of(undefined),
-    } as any);
-    component.entityType = TestEntity.ENTITY_TYPE;
-    const columnMapping: ColumnMapping = {
-      column: "test",
-      propertyName: "category",
+  it("should update the original column mapping and emit columnMappingChange", () => {
+    const originalColumnMapping: ColumnMapping = {
+      column: "Name",
+      propertyName: "test",
+    };
+    const newColumnMapping: ColumnMapping = {
+      column: "Name",
+      propertyName: "Test2",
     };
 
-    await component.openMappingComponent(columnMapping);
+    spyOn(component.columnMappingChange, "emit");
+
+    component.updateColumnMapping(originalColumnMapping, newColumnMapping);
 
-    expect(component.columnMappingChange.emit).toHaveBeenCalled();
+    expect(originalColumnMapping.propertyName).toBe("Test2");
+    expect(component.columnMappingChange.emit).toHaveBeenCalledWith([
+      ...component.columnMapping,
+    ]);
```

Comment:
turns out this is actually confirming that emit is called with an empty array ...
I fixed the expectation to be more useful.